### PR TITLE
TIP-856: Move Pim\BatchQueue Component and Bundle to Akeneo\Tool

### PR DESCRIPTION
**Pull request by arnolanglade**
**Description (for Contributor and Core Developer)**

A new name space has been created in this PR Akeneo\Tool which will contains our tools. All components and bundles from Akeneo\Xxx will be move to Akeneo\Tool\Xxx.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-3.0.md [hunk 1]
```diff
@@ -56,3 +56,5 @@
 - Move namespace  `Pim\Bundle\ApiBundle` to `Akeneo\Tool\Bundle\ApiBundle`
 - Move namespace `Pim\Component\Batch` to `Akeneo\Tool\Component\Batch`
 - Move namespace  `Pim\Bundle\BatchBundle` to `Akeneo\Tool\Bundle\BatchBundle`
+- Move namespace `Pim\Component\BatchQueue` to `Akeneo\Tool\Component\BatchQueue`
+- Move namespace  `Pim\Bundle\BatchQueueBundle` to `Akeneo\Tool\Bundle\BatchQueueBundle`
```
Comment (pierallard) [inline CHANGELOG-3.0.md]: still the spaces :)

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php [hunk 2]
```diff
@@ -121,7 +121,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
      * Return all the arguments of the command to execute.
      * Options are considered as arguments.
      *
-     * @param JobExecutionMessage $jobExecutionMessage
+     * @param \Akeneo\Tool\Component\BatchQueue\Queue\JobExecutionMessage $jobExecutionMessage
```
Comment (pierallard) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: use

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Hydrator/JobExecutionMessageHydrator.php [hunk 3]
```diff
@@ -41,7 +41,7 @@ public function __construct(EntityManagerInterface $entityManager)
      *
      * @throws MissingOptionsException
      *
-     * @return JobExecutionMessage
+     * @return \Akeneo\Tool\Component\BatchQueue\Queue\JobExecutionMessage
```
Comment (pierallard) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Hydrator/JobExecutionMessageHydrator.php]: ... etc

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Launcher/QueueJobLauncher.php [hunk 4]
```diff
@@ -50,13 +50,13 @@ class QueueJobLauncher implements JobLauncherInterface
...
-     * @param JobRegistry                $jobRegistry
-     * @param JobParametersValidator     $jobParametersValidator
-     * @param JobExecutionQueueInterface $queue
-     * @param string                     $environment
-     * @param EventDispatcherInterface  $eventDispatcher
+     * @param JobRepositoryInterface                                             $jobRepository
+     * @param JobParametersFactory                                               $jobParametersFactory
+     * @param JobRegistry                                                        $jobRegistry
+     * @param JobParametersValidator                                             $jobParametersValidator
+     * @param \Akeneo\Tool\Component\BatchQueue\Queue\JobExecutionQueueInterface $queue
```
Comment (pierallard) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Launcher/QueueJobLauncher.php]: ad libitum

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php (hunk 2 above)
Comment (ahocquard) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php]: already in the `use` so you can just keep it like before

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Hydrator/JobExecutionMessageHydrator.php (hunk 3 above)
Comment (ahocquard) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Hydrator/JobExecutionMessageHydrator.php]: same

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Launcher/QueueJobLauncher.php (hunk 4 above)
Comment (ahocquard) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Launcher/QueueJobLauncher.php]: same (it will revert all modification of this phpdoc btw)

**Pull request merged**